pytaxonkit.py

```diff
@@ -1112,10 +1113,17 @@ def lca(
         else:
             return None
     results = []
+    query_str_list = []
     for line in out.strip().split("\n"):
         queries, lcataxid = line.split("\t")
+        query_str_list.append(queries)
         results.append(int(lcataxid))
+    missing_ids = set(id_str_list) - set(query_str_list)
+    missing_indexes = [id_str_list.index(missing_id) for missing_id in missing_ids]
+    for missing_index in sorted(missing_indexes):
+        results.insert(missing_index, 0)
     if multi:
+        assert len(results) == len(ids)
```

Comment:
We can then also add this assert statement as now the length of the results will always be the same length as the input.